conda_self/cli/main_reset.py

```diff
@@ -20,38 +42,82 @@
     Reset the `base` environment to only the essential packages and plugins.
     """
 ).lstrip()
+SUCCESS_STATE = dedent(
+    """
+    SUCCESS!
+    Reset the `base` environment to {state} state.
+    """
+).lstrip()
 
 
 def configure_parser(parser: argparse.ArgumentParser) -> None:
     from conda.cli.helpers import add_output_and_prompt_options
 
     parser.description = HELP
     add_output_and_prompt_options(parser)
+    parser.add_argument(
+        "--snapshot",
+        choices=("current", "installer", "migrate"),
+        help=SNAPSHOT_HELP,
+    )
     parser.set_defaults(func=execute)
 
 
 def execute(args: argparse.Namespace) -> int:
     from conda.base.context import context
     from conda.reporters import confirm_yn
 
+    from ..constants import RESET_FILE_INSTALLER, RESET_FILE_MIGRATE
     from ..query import permanent_dependencies
     from ..reset import reset
 
     if not context.quiet:
         print(WHAT_TO_EXPECT)
 
-    confirm_yn(
-        "Proceed with resetting your 'base' environment?[y/n]:\n",
-        default="no",
-        dry_run=context.dry_run,
-    )
+    reset_data: dict[str, StateData] = {
+        "installer": {
+            "file_path": Path(sys.prefix, "conda-meta", RESET_FILE_INSTALLER),
+            "state_name": "installer-provided",
+        },
+        "migrate": {
+            "file_path": Path(sys.prefix, "conda-meta", RESET_FILE_MIGRATE),
+            "state_name": "post-migration",
+        },
+    }
+
+    reset_file: Path | None = None
+    state = ""
+    if not args.snapshot:
+        for state in ("migrate", "installer"):
+            if not reset_data[state]["file_path"].exists():
+                continue
+            reset_file = reset_data[state]["file_path"]
+            state = reset_data[state]["state_name"]
+            break
+    elif args.snapshot == "installer" or args.snapshot == "migrate":
+        reset_file = reset_data[args.snapshot]["file_path"]
+        state = reset_data[args.snapshot]["state_name"]
+
+    if reset_file and not reset_file.exists():
+        raise FileNotFoundError(
+            f"Failed to reset to `{args.snapshot}`. "
+            f"Required file {reset_file} not found."
+        )
+
+    prompt = "Proceed with resetting your 'base' environment"
```

Comment:
Is the name of the base environment here okay to hardcode?

Reply:
I think so since `conda-self` right now only resets the `base` environment. #75 is the place to remove the hardcoding, I think.